src/drt/src/pa/FlexPA.h

```diff
@@ -371,24 +371,33 @@ class FlexPA
       frInstTerm* inst_term);
 
   /**
-   * @brief Determines coordinates of an End Point given a Begin Point.
+   * @brief Generates an end_point given an begin_point in the direction
    *
-   * @param end_point the End Point to be filled
-   * @param layer_polys a vector with all the pin polygons
-   * @param begin_point the Begin Point
-   * @param layer_num the number of the layer where begin_point is
-   * @param dir the direction the End Point is from the Begin Point
-   * @param is_block if the instance is a macro block.
+   * @param layer_polys Pin Polygons on the layer (used for a check)
+   * TODO: maybe the check can be moves to isPointOusideShapes, but not sure
```

Comment:
I see that we now use only the first polygon in that vector. Maybe pass it directly instead of passing the vector?
Also I don't understand the TODO comment.

Reply:
The functions still checks for size, so I'm not sure only the first element can be passed.
The TODO comment is about the fact that now that the functions are split the `layer_polys` arguments only exists in `genEndPoint` to raise a warning. I though to moving this warning to `isPointOusideShapes` as it actually uses `layer_polys` for something. The problem is that `isPointOusideShapes` is only used in one of the previous cases, so I don't yet know if moving to it would matter, as then when `genEndPoint` is called alone it will not do this check